apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -428,40 +488,75 @@ public override void Prepare()
+            return ExecuteDbDataReaderImplAsync(
+                SpannerConnection,
+                GetExecuteSqlRequest(),
+                CommandTimeout,
+                GetSpannerTransaction(),
+                behavior,
+                null,
+                _synchronousCancellationTokenSource.Token)
                 .ResultWithUnwrappedExceptions();
         }
 
         /// <inheritdoc />
         protected override Task<DbDataReader> ExecuteDbDataReaderAsync(
             CommandBehavior behavior,
-            CancellationToken cancellationToken) => ExecuteDbDataReaderAsync(behavior, cancellationToken, null);
-
-        private async Task<DbDataReader> ExecuteDbDataReaderAsync(
+            CancellationToken cancellationToken) => ExecuteDbDataReaderImplAsync(
+                SpannerConnection,
+                GetExecuteSqlRequest(),
+                CommandTimeout,
+                GetSpannerTransaction(),
+                behavior,
+                null,
+                cancellationToken);
+
+        private static async Task<DbDataReader> ExecuteDbDataReaderImplAsync(
+            SpannerConnection spannerConnection,
+            ExecuteSqlRequest request,
+            int commandTimeout,
+            ISpannerTransaction spannerTransaction,
             CommandBehavior behavior,
-            CancellationToken cancellationToken,
-            SingleUseTransaction singleUseTransaction)
+            SingleUseTransaction singleUseTransaction,
+            CancellationToken cancellationToken)
         {
             // There must be a valid and open connection.
-            if (SpannerConnection == null)
+            if (spannerConnection == null)
             {
                 throw new InvalidOperationException(
                     "You must assign a SpannerConnection to this command to execute it.");
             }
 
-            if (SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Select)
+            if (!spannerConnection.IsOpen)
             {
-                throw new InvalidOperationException("You can only call ExecuteReader on a Select Command");
+                await spannerConnection.OpenAsync(cancellationToken).ConfigureAwait(false);
             }
 
-            if (!SpannerConnection.IsOpen)
+            if (!spannerConnection.IsOpen)
             {
-                await SpannerConnection.OpenAsync(cancellationToken).ConfigureAwait(false);
+                throw new InvalidOperationException("Unable to open the Spanner connection to the database.");
             }
 
-            if (!SpannerConnection.IsOpen)
+            var tx = singleUseTransaction ?? spannerTransaction;
+
+            // Execute the command.
+            var resultSet = await tx.ExecuteQueryAsync(request, cancellationToken)
+                .WithTimeout(TimeSpan.FromSeconds(commandTimeout), "The timeout of the SpannerCommand was exceeded.")
```

Comment:
actually this is a different timeout which is modifiable after the client is created (which is sort of a shame, but the design of ado.net).  So this means its possible for someone to set a timeout on the command which is different than the one on the client.
let's follow up on this after this change?  This change is altering the timeout, just ensuring it doesn't change in the middle of a call by snapping its state.